fastscape/tests/test_processes_tectonics.py

```diff
@@ -0,0 +1,163 @@
+import numpy as np
+
+import pytest
+
+from fastscape.processes import (BlockUplift,
+                                 HorizontalAdvection,
+                                 SurfaceAfterTectonics,
+                                 TectonicForcing,
+                                 TwoBlocksUplift)
+from fastscape.processes.context import FastscapelibContext
+
+
+def test_tectonic_forcing():
+    grid_shape = (3, 2)
+    uplift = np.random.uniform(size=grid_shape)
+    isostasy = np.random.uniform(size=grid_shape)
+    bedrock_advect = np.random.uniform(size=grid_shape)
+    surf_advect = np.random.uniform(size=grid_shape)
+    area = 300.
+    dt = 10.
+
+    p = TectonicForcing(bedrock_forcing_vars=[uplift, isostasy, bedrock_advect],
+                        surface_forcing_vars=[uplift, isostasy, surf_advect],
+                        grid_area=area)
+
+    p.run_step(dt)
+
+    sum_bedrock_forcing_vars = uplift + isostasy + bedrock_advect
+    np.testing.assert_equal(p.bedrock_upward, sum_bedrock_forcing_vars)
+
+    sum_surface_forcing_vars = uplift + isostasy + surf_advect
+    np.testing.assert_equal(p.surface_upward, sum_surface_forcing_vars)
+
+    domain_rate = np.sum(sum_surface_forcing_vars) * area / dt
+    assert p._domain_rate() == domain_rate
+
+
+def test_surface_after_tectonics():
+    grid_shape = (3, 2)
+    topo_elevation = np.random.uniform(size=grid_shape)
+    forced_motion = np.random.uniform(size=grid_shape)
+
+    p = SurfaceAfterTectonics(topo_elevation=topo_elevation,
+                              forced_motion=forced_motion)
+
+    p.run_step()
+
+    expected = topo_elevation + forced_motion
+    np.testing.assert_equal(p.elevation, expected)
+
+
+@pytest.mark.parametrize('b_status, expected_mask, expected_uplift', [
+    (np.array(['fixed_value', 'fixed_value', 'fixed_value', 'fixed_value']),
+    np.array([[0., 0., 0., 0.],
+              [0., 1., 1., 0.],
+              [0., 0., 0., 0.]]),
+    np.array([[0., 0., 0., 0.],
+              [0., 50., 50., 0.],
+              [0., 0., 0., 0.]])),
+    (np.array(['looped', 'looped', 'looped', 'looped']),
```

Comment:
Even if it works when taking `BlockUplift` separately, this is not a valid combination of `b_status` values if given to `BorderBoundary` (same the two other combinations below).

If we really want to cover all cases, we could use the following combinations:

```python
['fixed_value', 'fixed_value', 'fixed_value', 'fixed_value']
['core', 'fixed_value', 'fixed_value', 'fixed_value']
['fixed_value', 'core', 'fixed_value', 'fixed_value']
['fixed_value', 'fixed_value', 'core', 'fixed_value']
['fixed_value', 'fixed_value', 'fixed_value', 'core']
```